src/backend/joanie/core/factories.py

```diff
@@ -682,10 +685,54 @@ class Meta:
         OrderFactory,
         product__type=enums.PRODUCT_TYPE_CREDENTIAL,
     )
+    signed_on = None
 
     @factory.lazy_attribute
     def definition(self):
         """
         Return the order product contract definition.
         """
         return self.order.product.contract_definition
+
+    @factory.lazy_attribute
+    def context(self):
+        """
+        Lazily generate the contract context from the related order and contract definition.
+        """
+        if self.signed_on:
+            return {
+                "student": {
+                    "name": self.order.owner.get_full_name()
+                    or self.order.owner.username,
+                    # FIXME: address is not json serializable in definition_checksum
+                    # "address": self.order.owner.addresses.filter(is_main=True).first(),
```

Comment:
i didn't succeed to make this field work in checksum.
it may not be that important to have in on dev demo.

Reply:
You have to create a dict from the address object. Currently you are providing the reference to a `Model`.

Reply:
I used `address = AddressSerializer(user_address).data` as you advised me on slack.
... then it work's!